src/libraries/Common/src/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.cs

```diff
@@ -210,10 +210,10 @@ internal static partial class NetSecurityNative
             ref GssBuffer outBuffer);
 
         [LibraryImport(Interop.Libraries.NetSecurityNative, EntryPoint="NetSecurityNative_Unwrap")]
-        private static partial Status Unwrap(
+        private static unsafe partial Status Unwrap(
             out Status minorStatus,
             SafeGssContextHandle? contextHandle,
-            byte[] inputBytes,
+            byte* inputBytes,
```

Comment:
Do we need offset any more? It feels like if we pass pointer we can just do count.

Reply:
We don't. Unfortunately, dotnet/sqlclient uses the native APIs and so I didn't feel confident in changing it. I am quite sure there don't use this particular API though.

Reply:
It doesn't change the native code side, the prototype remains unchanged there. They basically copied the managed side of the interop.

Reply:
cc: @JRahnama  just in case.

Reply:
I cross-checked with [the SqlClient source](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/netcore/src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.cs) and this particular native method is not referenced so we can remove the `offset` parameter. I will do that in a follow-up PR (#71373) since I need to update the native interop there anyway.